src/parser/deserializer.ts

```diff
@@ -262,7 +330,9 @@ function deserializeObject(
         }
         arrayOptions['raw'] = true;
       }
-
+      if (!globalUTFValidation) {
+        arrayOptions = { ...options, validation: { utf8: shouldValidateKey } };
```

Comment:
```suggestion
        arrayOptions = { ...arrayOptions, validation: { utf8: shouldValidateKey } };
```
Just noticed there is some logic that modifies these, not sure if it really does anything, but better preserve it